frontends/ol-components/src/components/ChoiceBox/ChoiceBoxField.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react"
+import FormControl from "@mui/material/FormControl"
+import FormGroup from "@mui/material/FormGroup"
+import FormLabel from "@mui/material/FormLabel"
+import Grid, { type GridProps } from "@mui/material/Grid"
+import { ChoiceBox } from "./ChoiceBox"
+import type {
+  ChoiceBoxGridProps,
+  ChoiceBoxChoice,
+  ChoiceBoxProps,
+} from "./ChoiceBox"
+
+interface BaseChoiceBoxFieldProps extends ChoiceBoxGridProps {
+  label: React.ReactNode
+  choices: ChoiceBoxChoice[]
+  onChange: ChoiceBoxProps["onChange"]
+  className?: string
+}
+
+interface ChoiceBoxFieldProps extends BaseChoiceBoxFieldProps {
+  type: ChoiceBoxProps["type"]
+  name?: string
+  isChecked: (choice: ChoiceBoxChoice) => boolean
+}
+
+const ChoiceBoxField: React.FC<ChoiceBoxFieldProps> = ({
+  label,
+  name,
+  choices,
+  type,
+  isChecked,
+  onChange,
+  className,
+  gridProps,
+  gridItemProps,
+}: ChoiceBoxFieldProps) => {
+  const fieldGridProps: GridProps = {
+    spacing: "12px",
+    justifyContent: "center",
+    columns: {
+      lg: 12,
+      xs: 4,
+    },
+    ...gridProps,
+  }
+  const fieldGridItemProps = {
+    ...gridItemProps,
+  }
+  return (
+    <FormControl
+      className={className}
+      component="fieldset"
+      sx={{ width: "100%" }}
+    >
+      <FormLabel component="legend" sx={{ width: "100%" }}>
+        {label}
+      </FormLabel>
```

Comment:
A few overall comments here:

1. It would be nice to be able to support `required` and an error message.
2. I'm not sure what FormControl gives us. My understanding is that MUI uses this to wrap its inputs to manage some internal state. See https://mui.com/base-ui/react-form-control/. I chose not to use it for our TextField, and instead we have `FormFieldWrapper` in `FormHelpers`.

    - That said, FormFieldWrapper isn't directly directly usable here. It's expecting a single input, like select or text.
    - That said, the error message / description / required components in FormHelpers might be helpful.

3. I don't think we have very explicit component designs for `ChoiceBoxField`, but it seems to me it should match https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=4345-87576&m=dev :
    
    <img width="741" alt="Screenshot 2024-06-27 at 11 58 24 AM" src="https://github.com/mitodl/mit-open/assets/9010790/c38e21d1-1b70-4f2e-ad48-5ffee06c9575">

    I.e., the legend currently doesn't have the style we want.

    - We do have explicit field designs for text input. See https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=104-1582&m=dev

My inclination for actionable steps right now:
- Make title match profile page style (i.e., subtitle2, which is what the textfield and selectfield labels use, too)
- add spot for overall description + error message, using styles from FormHelpers. *You could alter FormFieldWrapper and use it, if you see fit. But the spacing between legend and input is also a little different, I think. 8px not 4px.*